hr_expense_cancel/models/hr_expense.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2019 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class HrExpenseSheet(models.Model):
+    _inherit = "hr.expense.sheet"
+
+    def action_cancel(self):
+        for sheet in self:
+            account_move = sheet.account_move_ids
+            sheet.account_move_ids = False
+            payments = sheet.payment_ids.filtered(lambda line: line.state != "cancel")
+            # case : cancel invoice from hr_expense
+            self._remove_reconcile_hr_invoice(account_move)
+            # If the sheet is paid then remove payments
+            if sheet.state in ("done", "approve"):
+                if sheet.expense_line_ids[:1].payment_mode == "own_account":
+                    self._remove_move_reconcile(payments, account_move)
+                    self._cancel_payments(payments)
+                else:
+                    # In this case, during the cancellation the journal entry
+                    # will be deleted
+                    self._cancel_payments(payments)
```

Comment:
We can remove the `else` block:
```suggestion
                if sheet.expense_line_ids[:1].payment_mode == "own_account":
                    self._remove_move_reconcile(payments, account_move)
                self._cancel_payments(payments)
```